public/store/dataset/actions.ts

```diff
@@ -102,31 +102,44 @@ export const actions = {
 			console.warn('`field` argument is missing');
 			return null;
 		}
+		const update: GeocodingPendingRequest = {
+			id: _.uniqueId(),
+			dataset: args.dataset,
+			type: DatasetPendingRequestType.GEOCODING,
+			field: args.field,
+			status: DatasetPendingRequestStatus.ERROR_REVIEWED,
+		};
+		mutations.updatePendingRequests(context, update);
 		return axios.post(`/distil/geocode/${args.dataset}/${args.field}`, {})
 			.then(() => {
-				// upon success pull the updated dataset, vars, and summaries
-				return Promise.all([
-					context.dispatch('fetchDataset', {
-						dataset: args.dataset
-					}),
-					context.dispatch('fetchVariables', {
-						dataset: args.dataset
-					}),
-					context.dispatch('fetchVariableSummary', {
-						dataset: args.dataset,
-						variable: GEOCODED_LON_PREFIX + args.field
-					}),
-					context.dispatch('fetchVariableSummary', {
-						dataset: args.dataset,
-						variable: GEOCODED_LAT_PREFIX + args.field
-					})
-				]);
+			mutations.updatePendingRequests(context, { ...update, status: DatasetPendingRequestStatus.RESOLVED });
 			})
 			.catch(error => {
+				mutations.updatePendingRequests(context, { ...update, status: DatasetPendingRequestStatus.ERROR });
 				console.error(error);
 			});
 	},
 
+	fetchDatasetAndVariables(context: DatasetContext, args: { dataset: string, field: string }) {
```

Comment:
Maybe rename this more specifically for the geocoding refresh, since it assumes the field is a geocoded variable.